sinonjs/samsam
# Upgrade ESLint and all dependencies

This PR upgrades ESLint and all related packages to latest versions.

#### Purpose (TL;DR)

In order to prepare for linting JSDoc, I thought it would be a good idea to get `samsam` caught up with the other projects.

That turned out to be more work than I anticipated 😮 

#### Solution

* Upgrade ESLint and related plugins and configs
* Fix lint violations
* Import `no-prototype-methods` from `sinon`
* Fix even more lint violations

#### How to verify - mandatory

1. Check out this branch
2. `npm install`
3. `npm run lint`
1. `npm test``
1. Observe that all is green

#### Checklist for author

- [x] References to standard library functions are [cached](https://github.com/sinonjs/sinon/pull/1523) ... **because ESLint now validates this** 🚀
